### ISSUE TITLE
Run migrations

### ISSUE THREAD

**Issue by antonkomarev** [laravel/ideas#1582]
It was started when I've faced a situation that Scrutinizer said that method name `up` is too short, and it's better to make it at least 3 characters long.

In the DocBlock of the `up` method we've got a text `Run the migrations`, so logically it could be renamed to `run` and then it will be clearly intuitive even without any documentation.
> What migration `run` method is doing?
> It is obviously running the migration!

I have heard a lot about risk of using `down` method and that it's not that great to revert the migrations, that migrations always should go only forward. Ideally `down` method should be removed in this scenario, but personally I found it useful while developing new features. When you create a new migration and apply it, you sometimes need to roll back to make changes and reapply. Because `down` was used as antonym for `up` it could be renamed to have more verbose name too, for example: `rollback`. Then `migrate:rollback` will be very clearly understood.

To have a support of old methods naming we could have a checks for existence of `up` & `down` methods.

**Comment by Patryk27** [comment]
So, basically, you want to introduce a major breaking change... because some tool said that `up` is too short name? That's not particularly convincing.

**Comment by antonkomarev** [comment]
No, I don't care what some external tools says, but I described why I started to look on this question. We had such type of breaking changes earlier. For example event `fire` method was renamed to `dispatch`. I'm thinking about it in context of naming consistency and intuitiveness of the name:
- we are running seeders with `run` method, but running migrations with `up` method;
- it will be more obvious that `artisan migrate:rollback` will execute `rollback` methods from migrations rather than `down`.

And I'm not proposing to make this changes right now, it's just a concept for the future.

**Comment by sisve** [comment]
Well, [Entity Framework uses Up/Down](https://www.learnentityframeworkcore.com/migrations), [Ruby on Rails uses up/down](https://edgeguides.rubyonrails.org/active_record_migrations.html), etc. It's a very common pattern, so it would be weird to step away from this established naming scheme.

Both methods are already optional. You can already decide to not implement down methods in your migrations.

> When you create a new migration and apply it, you sometimes need to roll back to make changes and reapply without need to seed all the application test data again. We could have configuration option to protect production environment from the migration rollback calls. Determine if rollback on production is allowed: config('database.production_rollback_allowed').

I don't see how reapplying the migrations will force you to run seeders. I often run migrations without seeding test data. Could you describe this scenario in detail?

Migration activities are already protected in production environments.

    **************************************
    *     Application In Production!     *
    **************************************
    
     Do you really wish to run this command? (yes/no) [no]:
     >

**Comment by ludo237** [comment]
- A table is up or down
- A service is up or down
- A database is up or down
- A migration is up or down

Being up, for migrations, means that it saved on the `migrations` table and it has been executed.

**Comment by antonkomarev** [comment]
> I don't see how reapplying the migrations will force you to run seeders. I often run migrations without seeding test data. Could you describe this scenario in detail?
I mean that if there will be no `rollback` method - the only possible way will be to make `migrate:fresh`, and then you will need to re-seed database. That was thinking out loud why reverting of the migration is important.

**Comment by sisve** [comment]
Can we agree that keeping the method names as-is is a good idea to match other languages and frameworks? If so, what other things does this issue want to accomplish?

**Comment by garygreen** [comment]
It's a down from me. See what I did there? 😊
